lib/workers/repository/update/pr/changelog/release-notes.ts

```diff
@@ -366,33 +366,45 @@ export async function getReleaseNotesMd(
             .replace(regEx(/^\s*#*\s*/), '')
             .split(' ')
             .filter(Boolean);
-          let body = section.replace(regEx(/.*?\n(-{3,}\n)?/), '').trim();
+          const body = section.replace(regEx(/.*?\n(-{3,}\n)?/), '').trim();
+          const notesSourceUrl = getNotesSourceUrl(
+            baseUrl,
+            repository,
+            project,
+            changelogFile,
+          );
+          const mdHeadingLink = title
+            .filter((word) => !isHttpUrl(word))
+            .join('-')
+            .replace(regEx(/[^A-Za-z0-9-]/g), '');
+          const url = `${notesSourceUrl}#${mdHeadingLink}`;
+          // Look for version in title
           for (const word of title) {
             if (word.includes(version) && !isHttpUrl(word)) {
               logger.trace({ body }, 'Found release notes for v' + version);
-              const notesSourceUrl = getNotesSourceUrl(
-                baseUrl,
-                repository,
-                project,
-                changelogFile,
-              );
-              const mdHeadingLink = title
-                .filter((word) => !isHttpUrl(word))
-                .join('-')
-                .replace(regEx(/[^A-Za-z0-9-]/g), '');
-              const url = `${notesSourceUrl}#${mdHeadingLink}`;
-              body = massageBody(body, baseUrl);
-              if (body?.length) {
-                try {
-                  body = await linkify(body, {
-                    repository: `${baseUrl}${repository}`,
-                  });
-                } catch (err) /* istanbul ignore next */ {
-                  logger.warn({ body, err }, 'linkify error');
-                }
-              }
               return {
-                body,
+                body: await linkifyBody(project, body),
+                url,
+                notesSourceUrl,
+              };
+            }
+          }
+          // Look for version in body - useful for monorepos. First check for "Releases (yyyy-mm-dd)" heading
+          const releasesRegex = regEx(/Releases ([0-9]{4}-[0-9]{2}-[0-9]{2})/);
+          if (packageName && heading.search(releasesRegex) !== -1) {
+            // Now check if any line contains both the package name and the version
+            const bodyLines = body.match(/^.*([\n\r]+|$)/gm) ?? [];
+            if (
+              bodyLines.some(
+                (line) =>
+                  line.includes(packageName) &&
+                  line.includes(version) &&
+                  !isHttpUrl(line),
+              )
+            ) {
```

Comment:
This is the extent of the new code.

If a line in the body contains the package name and the version we're searching for, it'll return the same `ChangeLogNotes` object (body + url + notesSourceUrl)